### PR TITLE
refactor: remove V2 and obsolete references

### PR DESCRIPTION
**Pull request by smol-ninja**
This PR closes 2nd and 3rd points of [Package tethering](https://github.com/sablier-labs/v2-core/issues/808).

### Changelog
- Remove obsolete references to Core
- Remove V2
- SVG contains "Sablier Lockup Linear" etc.

Screenshot 2024-07-30 at 14 49 13

### CODE REVIEW

File: src/core/SablierNFTDescriptor.sol [hunk 1]
```diff
@@ -347,14 +347,14 @@ contract SablierV2NFTDescriptor is ISablierV2NFTDescriptor {
     /// @dev Reverts if the symbol is unknown.
     function mapSymbol(IERC721Metadata sablier) internal view returns (string memory) {
         string memory symbol = sablier.symbol();
-        if (symbol.equal("SAB-V2-LOCKUP-LIN")) {
+        if (symbol.equal("SAB-LOCKUP-LIN") || symbol.equal("SAB-V2-LOCKUP-LIN")) {
```
Comment (smol-ninja) [inline src/core/SablierNFTDescriptor.sol:350]: Review required @sablier-labs/solidity.

File: src/core/SablierNFTDescriptor.sol [hunk 2]
```diff
@@ -52,7 +52,7 @@ contract SablierV2NFTDescriptor is ISablierV2NFTDescriptor {
         bool isTransferable;
         string json;
         bytes returnData;
-        ISablierV2Lockup sablier;
+        ISablierLockup sablier;
```
Comment (smol-ninja) [inline src/core/SablierNFTDescriptor.sol:55]: Should it be renamed as `lockup`?

**Review by smol-ninja — COMMENTED** [review]
@andreivladbrg tagging you for some review.

File: test/periphery/fork/merkle-lockup/MerkleLL.t.sol [hunk 3]
```diff
@@ -93,7 +93,14 @@ abstract contract MerkleLL_Fork_Test is Fork_Test {
 
         // Sort the leaves in ascending order to match the production environment.
         MerkleBuilder.sortLeaves(leaves);
-        vars.merkleRoot = getRoot(leaves.toBytes32());
+
+        // Compute the Merkle root.
+        if (leaves.toBytes32().length == 1) {
```
Comment (smol-ninja) [inline test/periphery/fork/merkle-lockup/MerkleLL.t.sol:98]: I had to add these conditions in the test because Murky requires the tree to have minimum 2 leaves.

File: src/core/SablierNFTDescriptor.sol (hunk 1 above)
Comment (andreivladbrg) [reply src/core/SablierNFTDescriptor.sol:350]: Good question,  and I also think that we should have compatible with previous versions

File: src/core/SablierNFTDescriptor.sol (hunk 2 above)
Comment (andreivladbrg) [reply src/core/SablierNFTDescriptor.sol:55]: no specific preferences, both work IMO

if we decide to rename it, we should for all:

```solidity
ISablierLockup sablier;
string sablierModel;
string sablierStringified;
```

File: README.md [hunk 4]
```diff
@@ -9,15 +9,15 @@
...
+This repository contains the core smart contracts of the Sablier Lockup Protocol. For higher-level logic, see the
 [sablier-labs/v2-periphery](https://github.com/sablier-labs/v2-periphery) repository.
 
 In-depth documentation is available at [docs.sablier.com](https://docs.sablier.com).
 
 ## Background
 
-Sablier V2 is a token distribution protocol used by DAOs and businesses for vesting, payroll, airdrops, and more. Our
-flagship model is the linear stream, which distributes assets on a continuous, by-the-second basis.
+Sablier Lockup Protocol is a token distribution protocol used by DAOs and businesses for vesting, payroll, airdrops, and
```
Comment (andreivladbrg) [inline README.md:19]: should we remove "payroll"

File: LICENSE.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ Parameters
 
 Licensor: Sablier Labs Ltd
 
-Licensed Work: Sablier V2 Core The Licensed Work is (C) 2024 Sablier Labs Ltd
+Licensed Work: Sablier Lockup The Licensed Work is (C) 2024 Sablier Labs Ltd
 
 Additional Use Grant: Any uses listed and defined at v2-core-license-grants.sablier.eth
```
Comment (andreivladbrg) [inline LICENSE.md:14]: @PaulRBerg should we take the ens domain for `lockup-license-grants.sablier.eth` ?

File: package.json [hunk 6]
```diff
@@ -38,10 +38,12 @@
     "cryptoassets",
     "ethereum",
     "foundry",
+    "lockup",
     "money-streaming",
     "real-time-finance",
     "sablier",
     "sablier-v2",
```
Comment (andreivladbrg) [inline package.json:45]: should we remove "sablier-v2"?

File: script/DeployDeterministicProtocol.s.sol [hunk 7]
```diff
@@ -1,43 +1,43 @@
...
+            SablierNFTDescriptor nftDescriptor,
+            SablierBatchLockup batchLockup,
+            SablierMerkleLockupFactory merkleLockupFactory
         )
     {
         bytes32 salt = constructCreate2Salt();
 
-        // Deploy V2 Core.
-        nftDescriptor = new SablierV2NFTDescriptor{ salt: salt }();
+        // Deploy  Core.
```
Comment (andreivladbrg) [inline script/DeployDeterministicProtocol.s.sol:31]: empty space

File: script/DeployDeterministicProtocol.s.sol [hunk 8]
```diff
@@ -1,43 +1,43 @@
...
+            new SablierLockupDynamic{ salt: salt }(initialAdmin, nftDescriptor, segmentCountMap[block.chainid]);
+        lockupLinear = new SablierLockupLinear{ salt: salt }(initialAdmin, nftDescriptor);
         lockupTranched =
-            new SablierV2LockupTranched{ salt: salt }(initialAdmin, nftDescriptor, trancheCountMap[block.chainid]);
+            new SablierLockupTranched{ salt: salt }(initialAdmin, nftDescriptor, trancheCountMap[block.chainid]);
 
-        // Deploy V2 Periphery.
-        batchLockup = new SablierV2BatchLockup{ salt: salt }();
-        merkleLockupFactory = new SablierV2MerkleLockupFactory{ salt: salt }();
+        // Deploy  Periphery.
```
Comment (andreivladbrg) [inline script/DeployDeterministicProtocol.s.sol:39]: empty space

File: script/DeployProtocol.s.sol [hunk 9]
```diff
@@ -1,39 +1,39 @@
...
+            SablierBatchLockup batchLockup,
+            SablierMerkleLockupFactory merkleLockupFactory
         )
     {
-        // Deploy V2 Core.
-        nftDescriptor = new SablierV2NFTDescriptor();
-        lockupDynamic = new SablierV2LockupDynamic(initialAdmin, nftDescriptor, segmentCountMap[block.chainid]);
-        lockupLinear = new SablierV2LockupLinear(initialAdmin, nftDescriptor);
-        lockupTranched = new SablierV2LockupTranched(initialAdmin, nftDescriptor, trancheCountMap[block.chainid]);
+        // Deploy  Core.
```
Comment (andreivladbrg) [inline script/DeployProtocol.s.sol:29]: empty space

File: script/DeployProtocol.s.sol [hunk 10]
```diff
@@ -1,39 +1,39 @@
...
+        nftDescriptor = new SablierNFTDescriptor();
+        lockupDynamic = new SablierLockupDynamic(initialAdmin, nftDescriptor, segmentCountMap[block.chainid]);
+        lockupLinear = new SablierLockupLinear(initialAdmin, nftDescriptor);
+        lockupTranched = new SablierLockupTranched(initialAdmin, nftDescriptor, trancheCountMap[block.chainid]);
 
-        // Deploy V2 Periphery.
-        batchLockup = new SablierV2BatchLockup();
-        merkleLockupFactory = new SablierV2MerkleLockupFactory();
+        // Deploy  Periphery.
```
Comment (andreivladbrg) [inline script/DeployProtocol.s.sol:35]: empty space

File: script/periphery/CreateMerkleLL.s.sol [hunk 11]
```diff
@@ -3,20 +3,20 @@ pragma solidity >=0.8.22 <0.9.0;
...
 
 contract CreateMerkleLL is BaseScript {
     /// @dev Deploy via Forge.
-    function run() public virtual broadcast returns (ISablierV2MerkleLL merkleLL) {
+    function run() public virtual broadcast returns (ISablierMerkleLL merkleLL) {
         // Prepare the constructor parameters.
-        ISablierV2MerkleLockupFactory merkleLockupFactory =
-            ISablierV2MerkleLockupFactory(0xF35aB407CF28012Ba57CAF5ee2f6d6E4420253bc);
+        ISablierMerkleLockupFactory merkleLockupFactory =
+            ISablierMerkleLockupFactory(0xF35aB407CF28012Ba57CAF5ee2f6d6E4420253bc);
```
Comment (andreivladbrg) [inline script/periphery/CreateMerkleLL.s.sol:19]: let's add a TODO, to udpate these once deployed new contracts so that we don't forget

File: script/periphery/CreateMerkleLL.s.sol [hunk 12]
```diff
@@ -28,7 +28,7 @@ contract CreateMerkleLL is BaseScript {
         baseParams.name = "The Boys LL";
         baseParams.transferable = true;
 
-        ISablierV2LockupLinear lockupLinear = ISablierV2LockupLinear(0x3962f6585946823440d274aD7C719B02b49DE51E);
+        ISablierLockupLinear lockupLinear = ISablierLockupLinear(0x3962f6585946823440d274aD7C719B02b49DE51E);
```
Comment (andreivladbrg) [inline script/periphery/CreateMerkleLL.s.sol:31]: same as above

File: script/periphery/CreateMerkleLT.s.sol [hunk 13]
```diff
@@ -4,19 +4,19 @@ pragma solidity >=0.8.22 <0.9.0;
...
 
 contract CreateMerkleLT is BaseScript {
     /// @dev Deploy via Forge.
-    function run() public virtual broadcast returns (ISablierV2MerkleLT merkleLT) {
+    function run() public virtual broadcast returns (ISablierMerkleLT merkleLT) {
         // Prepare the constructor parameters.
-        ISablierV2MerkleLockupFactory merkleLockupFactory =
-            ISablierV2MerkleLockupFactory(0xF35aB407CF28012Ba57CAF5ee2f6d6E4420253bc);
+        ISablierMerkleLockupFactory merkleLockupFactory =
+            ISablierMerkleLockupFactory(0xF35aB407CF28012Ba57CAF5ee2f6d6E4420253bc);
```
Comment (andreivladbrg) [inline script/periphery/CreateMerkleLT.s.sol:19]: same as in `CreateMerkleLL`

File: src/core/SablierLockupDynamic.sol [hunk 14]
```diff
@@ -8,20 +8,20 @@ import { PRBMathCastingUint128 as CastingUint128 } from "@prb/math/src/casting/U
...
-███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝    ██║   ██║ █████╔╝
-╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗    ╚██╗ ██╔╝██╔═══╝
-███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║     ╚████╔╝ ███████╗
-╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝      ╚═══╝  ╚══════╝
+███████╗ █████╗ ██████╗ ██╗     ██╗███████╗██████╗
+██╔════╝██╔══██╗██╔══██╗██║     ██║██╔════╝██╔══██╗
+███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝
+╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗
+███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║
+╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝
```
Comment (andreivladbrg) [inline src/core/SablierLockupDynamic.sol:24]: let's move "lockup" in the first row

File: src/core/SablierLockupLinear.sol [hunk 15]
```diff
@@ -6,21 +6,21 @@ import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.s
...
-██╔════╝██╔══██╗██╔══██╗██║     ██║██╔════╝██╔══██╗    ██║   ██║╚════██╗
-███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝    ██║   ██║ █████╔╝
-╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗    ╚██╗ ██╔╝██╔═══╝
-███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║     ╚████╔╝ ███████╗
-╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝      ╚═══╝  ╚══════╝
+███████╗ █████╗ ██████╗ ██╗     ██╗███████╗██████╗
+██╔════╝██╔══██╗██╔══██╗██║     ██║██╔════╝██╔══██╗
+███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝
+╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗
+███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║
```
Comment (andreivladbrg) [inline src/core/SablierLockupLinear.sol:22]: same as in dynamic

File: src/core/SablierLockupTranched.sol [hunk 16]
```diff
@@ -5,20 +5,20 @@ import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
...
-██╔════╝██╔══██╗██╔══██╗██║     ██║██╔════╝██╔══██╗    ██║   ██║╚════██╗
-███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝    ██║   ██║ █████╔╝
-╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗    ╚██╗ ██╔╝██╔═══╝
-███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║     ╚████╔╝ ███████╗
-╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝      ╚═══╝  ╚══════╝
+███████╗ █████╗ ██████╗ ██╗     ██╗███████╗██████╗
+██╔════╝██╔══██╗██╔══██╗██║     ██║██╔════╝██╔══██╗
+███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝
+╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗
+███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║
```
Comment (andreivladbrg) [inline src/core/SablierLockupTranched.sol:20]: same as in dynamic

File: src/core/SablierNFTDescriptor.sol [hunk 17]
```diff
@@ -17,12 +17,12 @@ import { SVGElements } from "./libraries/SVGElements.sol";
...
-███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝    ██║   ██║ █████╔╝
-╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗    ╚██╗ ██╔╝██╔═══╝
-███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║     ╚████╔╝ ███████╗
-╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝      ╚═══╝  ╚══════╝
+███████╗ █████╗ ██████╗ ██╗     ██╗███████╗██████╗
+██╔════╝██╔══██╗██╔══██╗██║     ██║██╔════╝██╔══██╗
+███████╗███████║██████╔╝██║     ██║█████╗  ██████╔╝
+╚════██║██╔══██║██╔══██╗██║     ██║██╔══╝  ██╔══██╗
+███████║██║  ██║██████╔╝███████╗██║███████╗██║  ██║
+╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝
```
Comment (andreivladbrg) [inline src/core/SablierNFTDescriptor.sol:25]: sablier nft in the first row

File: src/core/libraries/Errors.sol [hunk 18]
```diff
@@ -18,142 +18,140 @@ library Errors {
     error DelegateCall();
 
     /*//////////////////////////////////////////////////////////////////////////
-                                 SABLIER-V2-LOCKUP
+                                 SABLIER-LOCKUP
```
Comment (andreivladbrg) [inline src/core/libraries/Errors.sol:21]: not centered

we are using [this](https://github.com/transmissions11/headers) for the headers, but with 76 spaces instead of 64

File: src/periphery/SablierBatchLockup.sol [hunk 19]
```diff
@@ -4,27 +4,27 @@ pragma solidity >=0.8.22;
...
-/// @notice See the documentation in {ISablierV2BatchLockup}.
-contract SablierV2BatchLockup is ISablierV2BatchLockup {
+/// @title SablierBatchLockup
+/// @notice See the documentation in {ISablierBatchLockup}.
+contract SablierBatchLockup is ISablierBatchLockup {
     using SafeERC20 for IERC20;
 
     /*//////////////////////////////////////////////////////////////////////////
-                             SABLIER-V2-LOCKUP-DYNAMIC
+                             SABLIER-LOCKUP-DYNAMIC
```
Comment (andreivladbrg) [inline src/periphery/SablierBatchLockup.sol:22]: not centered

File: src/periphery/interfaces/ISablierBatchLockup.sol [hunk 20]
```diff
@@ -3,32 +3,32 @@ pragma solidity >=0.8.22;
...
 
-/// @title ISablierV2BatchLockup
-/// @notice Helper to batch create Sablier V2 Lockup streams.
-interface ISablierV2BatchLockup {
+/// @title ISablierBatchLockup
+/// @notice Helper to batch create Lockup streams.
+interface ISablierBatchLockup {
     /*//////////////////////////////////////////////////////////////////////////
-                              SABLIER-V2-LOCKUP-LINEAR
+                              SABLIER-LOCKUP-LINEAR
```
Comment (andreivladbrg) [inline src/periphery/interfaces/ISablierBatchLockup.sol:16]: not centered, and not alphabetically ordered (just realized)

File: test/core/fork/NFTDescriptor.t.sol [hunk 21]
```diff
@@ -28,15 +28,15 @@ contract NFTDescriptor_Fork_Test is Fork_Test {
...
     modifier loadDeployments_V2_0() {
-        lockupDynamic = ISablierV2LockupDynamic(0x39EFdC3dbB57B2388CcC4bb40aC4CB1226Bc9E44);
-        lockupLinear = ISablierV2LockupLinear(0xB10daee1FCF62243aE27776D7a92D39dC8740f95);
+        lockupDynamic = ISablierLockupDynamic(0x39EFdC3dbB57B2388CcC4bb40aC4CB1226Bc9E44);
+        lockupLinear = ISablierLockupLinear(0xB10daee1FCF62243aE27776D7a92D39dC8740f95);
         _;
     }
 
     /// @dev Loads the Lockup V2.1 contracts pre-deployed on Mainnet.
     modifier loadDeployments_V2_1() {
```
Comment (andreivladbrg) [inline test/core/fork/NFTDescriptor.t.sol:37]: let's change the `V2.1` to package version, i.e. `v1.1.2`

File: src/core/SablierNFTDescriptor.sol (hunk 1 above)
Comment (smol-ninja) [reply src/core/SablierNFTDescriptor.sol:350]: Sounds good. Thanks for confirming.

**Review by andreivladbrg — CHANGES_REQUESTED** [review]
Great work!

I see that there are multiple references , especially`.MD` files, that point to `v2-core`. Since we will most likely coordinate the merge of `staging` to `main` and the renaming of the repo to `lockup`, I think it's safe to change them now, wdyt? 
For example:

https://github.com/sablier-labs/v2-core/blob/a34e5287bb77c3e20b8fb9d3ef349a4a80d990aa/foundry.toml#L89

Feedback below, which I will create a PR with the some of the fixes:

File: src/core/SablierNFTDescriptor.sol (hunk 2 above)
Comment (smol-ninja) [reply src/core/SablierNFTDescriptor.sol:55]: Forgot to add a point that the choice of `sablier` vs `lockup` depends on the fact that whether the same NFT descriptor would be used for Lockup as well as Flow. If both, then `sablier` would be the right choice and if not, then `lockup` would be the right choice.

File: test/periphery/fork/merkle-lockup/MerkleLL.t.sol (hunk 3 above)
Comment (andreivladbrg) [reply test/periphery/fork/merkle-lockup/MerkleLL.t.sol:98]: let's remove them

it is a mistake introduced here: 

https://github.com/sablier-labs/v2-core/blob/a97e5ae278e82a91533d42ae4df2cd3f95c14f81/test/periphery/fork/merkle-lockup/MerkleLL.t.sol#L61

it should be: `vm.assume(params.leafData.length > 1);`

commit when was introduced: https://github.com/sablier-labs/v2-core/commit/a97e5ae278e82a91533d42ae4df2cd3f95c14f81

Comment (smol-ninja) [reply test/periphery/fork/merkle-lockup/MerkleLL.t.sol:98]: This is why I introduced that change https://github.com/sablier-labs/v2-periphery/issues/266

Comment (andreivladbrg) [reply test/periphery/fork/merkle-lockup/MerkleLL.t.sol:98]: completely missed it, sorry

let's leave it then like this

File: src/core/SablierNFTDescriptor.sol (hunk 2 above)
Comment (smol-ninja) [reply src/core/SablierNFTDescriptor.sol]: https://github.com/sablier-labs/v2-core/pull/994/commits/513843469ebc4e746045fad2712f975b452965ff

File: README.md (hunk 4 above)
Comment (smol-ninja) [reply README.md:19]: Not sure. Lockup can be used for payroll as well 😅

File: package.json (hunk 6 above)
Comment (smol-ninja) [reply package.json:45]: I left it intentionally. These are search keywords so having "sablier-v2" would result into this repo which is a good thing.

File: script/periphery/CreateMerkleLL.s.sol (hunk 11 above)
Comment (smol-ninja) [reply script/periphery/CreateMerkleLL.s.sol]: Should we add TODO in Fork.t.sol as well?

File: src/core/libraries/Errors.sol (hunk 18 above)
Comment (smol-ninja) [reply src/core/libraries/Errors.sol]: Thanks for the tip.

File: test/core/fork/NFTDescriptor.t.sol (hunk 21 above)
Comment (smol-ninja) [reply test/core/fork/NFTDescriptor.t.sol:37]: Sounds good.

**Comment by smol-ninja** [comment]
> I see that there are multiple references , especially.MD files, that point to v2-core. Since we will most likely coordinate the merge of staging to main and the renaming of the repo to lockup, I think it's safe to change them now, wdyt?
For example:

Yes, I was keeping them until we rename the repo but it does not matter. So, I've changed them in https://github.com/sablier-labs/v2-core/pull/994/commits/5ad448164fff83588dee602e8c8f854247ca8c38.

**Comment by smol-ninja** [comment]
Added a new commit: https://github.com/sablier-labs/v2-core/pull/994/commits/11cd564bc419847ba42fddd0bcfb0b2b7a7b5c64. LMK if it all looks good now @andreivladbrg.

PS: there are a few open conversations pending your reply.

File: README.md (hunk 4 above)
Comment (andreivladbrg) [reply README.md:19]: agree

**Review by andreivladbrg — APPROVED** [review]
pushed a new commit c6d00517912de9405f7f2afa9d0050e62154a34e

lmk if it looks good

File: LICENSE.md (hunk 5 above)
Comment (PaulRBerg) [reply LICENSE.md:14]: Yes, we should.

For anything that requires submitting a multisig transaction, you can tag @maxdesalle. He initiates the multisig txs — I just execute them.
